src/Mobs/Path.cpp

```diff
@@ -0,0 +1,382 @@
+	{
+		FinishCalculation();
+	}
+}
+
+
+
+
+
+void cPath::FinishCalculation()
+{
+	for (auto&& pair : m_Map)
+	{
+		delete pair.second;
+	}
+	
+	m_Map.clear();
+	m_OpenList.empty();
+}
+
+
+
+
+
+void cPath::FinishCalculation(ePathFinderStatus a_NewStatus)
+{
+	m_Status = a_NewStatus;
+	FinishCalculation();
+}
+
+
+
+
+
+cPathCell * cPath::GetCell(const Vector3d & a_Location)
+{
+	// Create the cell in the hash table if it's not already there.
+	cPathCell * Cell;
+	if (m_Map.count(a_Location) == 0)  // Case 1: Cell is not on any list. We've never checked this cell before.
+	{
+		Cell = new cPathCell();
+		Cell->m_Location = a_Location;
+		m_Map[a_Location] = Cell;
+		Cell->m_IsSolid = IsSolid(a_Location);
+		Cell->m_Status = eCellStatus::NOLIST;
+		#ifdef COMPILING_PATHFIND_DEBUGGER
+		#ifdef COMPILING_PATHFIND_DEBUGGER_MARK_UNCHECKED
+		si::setBlock(a_Location.x, a_Location.y, a_Location.z, debug_unchecked, Cell->m_IsSolid ? NORMAL : MINI);
+		#endif
+		#endif
+		return Cell;
+	}
+	else return m_Map[a_Location];
+}
+
+
+
+
+
+void cPath::ProcessCell(cPathCell * a_Cell, cPathCell * a_Caller, int a_GDelta)
+{
+	// Case 1: Cell is in the closed list, ignore it.
+	if (a_Cell->m_Status == eCellStatus::CLOSEDLIST)
+	{
+		return;
+	}
+	if (a_Cell->m_Status == eCellStatus::NOLIST)  // Case 2: The cell is not in any list.
+	{
+		// Cell is walkable, add it to the open list.
+		// Note that non-walkable cells are filtered out in Step_internal();
+		// Special case: Start cell goes here, gDelta is 0, caller is NULL.
+		a_Cell->m_Parent = a_Caller;
+		if (a_Caller != NULL)
+		{
+			a_Cell->m_G = a_Caller->m_G + a_GDelta;
+		}
+		else
+		{
+			a_Cell->m_G = 0;
+		}
+		// Calculate H. This is A*'s Heuristics value.
+		#if DISTANCE_MANHATTEN == 1
+		// Manhatten distance. DeltaX + DeltaY + DeltaZ.
+		a_Cell->m_H = 10 * (abs(a_Cell->m_Location.x-m_Destination.x) + abs(a_Cell->m_Location.y-m_Destination.y) + abs(a_Cell->m_Location.z-m_Destination.z));
+		#else
+		// Euclidian distance. sqrt(DeltaX^2 + DeltaY^2 + DeltaZ^2), more precise.
+		a_Cell->m_H = std::sqrt( (a_Cell->m_Location.x-m_Destination.x) * (a_Cell->m_Location.x-m_Destination.x) * 100+ (a_Cell->m_Location.y-m_Destination.y) * (a_Cell->m_Location.y-m_Destination.y) * 100 + (a_Cell->m_Location.z-m_Destination.z) * (a_Cell->m_Location.z-m_Destination.z) * 100);
+		#endif
+		#if HEURISTICS_ONLY == 1
+		a_Cell->m_F = a_Cell->m_H;  // Depth-first search(Might be the wrong name). Faster, can yeild paths that are far from optimal.
+		#else
+		a_Cell->m_F = a_Cell->m_H + a_Cell->m_G;  // Regular A*.
+		#endif
+		OpenListAdd(a_Cell);
+		return;
+	}
+	
+	// Case 3: Cell is in the open list, check if G and H need an update.
+	int NewG = a_Caller->m_G + a_GDelta;
+	if (NewG < a_Cell->m_G)
+	{
+		a_Cell->m_G = NewG;
+		a_Cell->m_H = a_Cell->m_F + a_Cell->m_G;
+		a_Cell->m_Parent = a_Caller;
+	}
+	
+}
+
+
+
+
+
+ePathFinderStatus cPath::Step()
+{
+	if (m_Status == ePathFinderStatus::CALCULATING)
```

Comment:
Invert the condition and return early:
```
if (m_Status != CALCULATING)
{
    return m_Status;
}
```
You'll have one less indent level, much easier to read